tests/Doctrine/Tests/DBAL/Functional/ExceptionTest.php

```diff
@@ -87,27 +86,13 @@ public function testForeignKeyConstraintViolationExceptionOnInsert() : void
         try {
             $this->connection->insert('constraint_error_table', ['id' => 1]);
             $this->connection->insert('owning_table', ['id' => 1, 'constraint_id' => 1]);
-        } catch (Throwable $exception) {
-            $this->tearDownForeignKeyConstraintViolationExceptionTest();
-
-            throw $exception;
-        }
 
-        $this->expectException(Exception\ForeignKeyConstraintViolationException::class);
+            $this->expectException(Exception\ForeignKeyConstraintViolationException::class);
 
-        try {
             $this->connection->insert('owning_table', ['id' => 2, 'constraint_id' => 2]);
-        } catch (Exception\ForeignKeyConstraintViolationException $exception) {
-            $this->tearDownForeignKeyConstraintViolationExceptionTest();
-
-            throw $exception;
-        } catch (Throwable $exception) {
+        } finally {
             $this->tearDownForeignKeyConstraintViolationExceptionTest();
```

Comment:
Would it make more sense to extract this test to a separate class and let PHPUnit take care of calling it? Or just make this routine part of the shared teardown? Calling it manually and having flow control logic in a test still smells.

Reply:
It does smell, but smells less than before! ;-) That's a good point, though. Moved to their own class with proper `setUp()/tearDown()` in da3a944e8451f0e543a622ed27fd4c20badbf7fb.